docs/vcg-debugging.md

```diff
@@ -0,0 +1,124 @@
+<!--
+     Copyright 2024, Proofcraft Pty Ltd
+
+     SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+# VCG debugging techniques for CRefine
+
+## Unprovable goals and `False` in the post condition
+
+When using `vcg`, situations arise where schematic instantiation to `False` can
+happen due to unification failure -- mostly during simplification. This is
+usually hard to find because `False` itself does not necessarily show up, for
+example you may get something like this:
+
+```isabelle
+   ∀cte cap.
+      ccap_relation (cteCap cte) cap ⟶
+      cap_get_tag cap ≠ signed cap_vspace_cap ∧ ...
+```
+
+Which is obviously not provable, because we don't have any information about
+`cap` to work with. What it should have generated is something like:
+
+```isabelle
+   ∀cte cap.
+       ccap_relation (cteCap cte) cap ⟶
+       (∀t. cap_get_tag cap = signed cap_vspace_cap ⟶ ...)
+```
+
+This means that some conjunct in the `...` above got resolved to `False`, which
+got simplified to the unsolvable goal.
+
+The way to track down these issues is to go over the `vcg` steps to see which
+ones introduced the unexpected result. However, with large goals this is very
+difficult to work with. What else can help?
+
+First, try removing `(P ⟶ False) = (¬ P)` from the simpset, so we can look for
+`False` in the goals:
+
+```isabelle
+supply simp_thms(19)[simp del]
```

Comment:
you sometimes indent and sometimes not, not sure if markdown cares

Reply:
I should indent, markdown does show the difference